File: example/python/streamplot_demo/streamplot_demo.py
#!/usr/bin/env python3
"""Streamplot demo - demonstrates streamline plotting"""

import numpy as np
import sys

if "--matplotlib" in sys.argv:
    import matplotlib.pyplot as plt
    backend = "matplotlib"
else:
    import fortplot.fortplot as plt
    backend = "fortplot"

def main():
    nx, ny = 20, 20

    # Create grid
    x = np.linspace(-2.0, 2.0, nx)
    y = np.linspace(-2.0, 2.0, ny)
    X, Y = np.meshgrid(x, y)

    # Create circular flow field
    U = -Y
    V = X

    # Create figure and add streamplot
    plt.figure(figsize=(10, 7.5))
    plt.streamplot(X, Y, U, V, density=1.0)
    plt.xlabel('X')
    plt.ylabel('Y')
    plt.title('Streamline Plot Demo - Circular Flow')
    plt.savefig('streamplot_demo.png')
    plt.savefig('streamplot_demo.pdf')

    # Export TXT for fortplot mode
    if backend == "fortplot":
        with open('streamplot_demo.txt', 'w') as f:
            f.write("Streamline Plot Demo - Circular Flow\n")
            f.write("=====================================\n\n")
            f.write("Grid size: {}x{}\n".format(nx, ny))
            f.write("X range: -2.0 to 2.0\n")
            f.write("Y range: -2.0 to 2.0\n")
            f.write("Flow field: Circular (U=-Y, V=X)\n")
            f.write("Density: 1.0\n")

    print('Streamplot demo completed!')

if __name__ == "__main__":
    main()

File: example/fortran/streamplot_demo/streamplot_demo.f90
program streamplot_demo
    use, intrinsic :: iso_fortran_env, only: wp => real64
    use fortplot, only: figure, streamplot, xlabel, ylabel, title, savefig
    implicit none

    integer, parameter :: nx = 20, ny = 20
    real(wp), dimension(nx) :: x
    real(wp), dimension(ny) :: y
    real(wp), dimension(nx, ny) :: u, v
    integer :: i, j

    ! Create grid
    do i = 1, nx
        x(i) = -2.0_wp + 4.0_wp * real(i-1, wp) / real(nx-1, wp)
    end do

    do j = 1, ny
        y(j) = -2.0_wp + 4.0_wp * real(j-1, wp) / real(ny-1, wp)
    end do

    ! Create circular flow field
    do j = 1, ny
        do i = 1, nx
            u(i,j) = -y(j)
            v(i,j) = x(i)
        end do
    end do

    ! Base streamline plot (no arrows)
    call figure(figsize=[8.0_wp, 6.0_wp])
    call streamplot(x, y, u, v, density=1.0_wp)
    call xlabel('X')
    call ylabel('Y')
    call title('Streamline Plot Demo - Circular Flow')

    ! Save base figure
    call savefig('output/example/fortran/streamplot_demo/streamplot_demo.png')
    call savefig('output/example/fortran/streamplot_demo/streamplot_demo.pdf')
    call savefig('output/example/fortran/streamplot_demo/streamplot_demo.txt')

    ! Arrow variant: emphasize direction with arrowheads
    call figure(figsize=[8.0_wp, 6.0_wp])
    call streamplot(x, y, u, v, density=1.0_wp, arrowsize=1.5_wp, arrowstyle='->')
    call xlabel('X')
    call ylabel('Y')
    call title('Streamline Plot Demo - With Arrows')

    ! Save arrow variant
    call savefig('output/example/fortran/streamplot_demo/streamplot_arrows.png')
    call savefig('output/example/fortran/streamplot_demo/streamplot_arrows.pdf')
    call savefig('output/example/fortran/streamplot_demo/streamplot_arrows.txt')

    print *, 'Streamplot demo completed!'

end program streamplot_demo
